polars/polars-io/src/ipc.rs

```diff
@@ -161,6 +164,9 @@ where
                 let i = schema.index_of(column)?;
                 prj.push(i)
             }
+
+            // Ipc reader panics if the projection is not in increasing order, so sorting is the safer way.
+            prj.sort();
```

Comment:
Maybe also remove duplicates after sorting:
https://github.com/jorgecarleitao/arrow2/blob/3caa20421f8a6950dec6e2ad1c54125ef3a896ab/src/io/ipc/read/reader.rs#L236

Reply:
Sure, we can remove duplicates, but It seems that it would be more helpful to return an error if there is a duplicate.

It's not good to get an error in `select C, B`, but wouldn't it be better to get an error in `select B, B`?

Reply:
I prefer returning an error indeed. The API expects unique columns.